src/Features/CSharp/Portable/SignatureHelp/ConstructorInitializerSignatureHelpProvider.cs

```diff
@@ -82,9 +82,11 @@ private static bool IsArgumentListToken(ConstructorInitializerSyntax expression,
                 return null;
             }
 
+            var currentConstructor = semanticModel.GetDeclaredSymbol(constructorInitializer.Parent);
+
             var symbolDisplayService = document.Project.LanguageServices.GetService<ISymbolDisplayService>();
             var accessibleConstructors = type.InstanceConstructors
-                                             .WhereAsArray(c => c.IsAccessibleWithin(within))
+                                             .WhereAsArray(c => c.IsAccessibleWithin(within) && c != currentConstructor)
```

Comment:
Nit: You should always use `.Equals` for ISymbol. They aren't guaranteed to have reference equality.